src/LoginWithAmazon.test.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import TestUtils from 'react';
+import { shallow, mount } from 'enzyme';
+import LoginWithAmazon from './LoginWithAmazon';
+
+let loginWithAmazon;
+let loginWithAmazonInstance;
+let mockSetAccessToken;
+let authorize;
+
+beforeEach(() => {
+    mockSetAccessToken = jest.fn();
+    loginWithAmazon = shallow(<LoginWithAmazon setAccessToken={mockSetAccessToken}/>);
+    loginWithAmazonInstance = loginWithAmazon.instance();
+    authorize = jest.fn();
+
+    Object.defineProperty(window, 'amazon', {
+        value: { Login: { authorize: authorize } },
+        configurable: true
+    });
+});
+
+it('renders without crashing', () => {
+    loginWithAmazon.find('button').simulate('click');
+
+    // Verify authorize called once
+    expect(authorize.mock.calls.length).toBe(1);
+});
+
+it('request access code errors out', () => {
```

Comment:
jest's describe / it convention is meant to be read as a sentence. This just says what scenario is being tested. 

It should be something like - 
it('logs an error when we encounter a failure trying to retrieve access_code')